Curs3/app/src/main/java/com/example/curs3/MainActivity.java

```diff
@@ -0,0 +1,133 @@
+package com.example.curs3;
+
+import android.os.Bundle;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+    private EditText email;
+    private EditText phone;
+    private CheckBox terms;
+    private Button submit;
+    private TextView user_input;
+
+    private Button next_layout_1_2;
+    private Button next_layout_2_3;
+
+    private Spinner androidVersions;
+    private ArrayList<String> androidVersionsList;
+    private ArrayAdapter<String> spinnerAdapter;
+
+    protected void initSubmitButton() {
+        submit.setOnClickListener(v -> {
+            boolean isValidInput = true;
+
+            String emailStr = email.getText().toString();
+            String phoneStr = phone.getText().toString();
+
+            if (emailStr.isEmpty() ||
+                    !Patterns.EMAIL_ADDRESS.matcher(emailStr).matches()) {
+                email.setError("Fill the input with a valid email address");
+                isValidInput = false;
+            }
+
+            if (phoneStr.isEmpty() ||
+                    !Patterns.PHONE.matcher(phoneStr).matches()) {
+                phone.setError("Fill the input with a valid phone number");
+                isValidInput = false;
+            }
+
+            if (!terms.isChecked()) {
+                terms.setError("You must accept the terms and conditions");
+                isValidInput = false;
+            }
+
+            String userInputStr = "email: " + emailStr +
+                    ", phone: " + phoneStr +
+                    ", terms agreed: " + terms.isChecked();
+            user_input.setText(userInputStr);
+
+            if (isValidInput) {
+                setContentView(R.layout.code_challenge_2_1);
+
+                next_layout_1_2 = findViewById(R.id.button_next_layout_1_2);
+
+                initNextLayoutButton12();
+            }
+        });
+    }
+
+    protected void initNextLayoutButton12() {
+        next_layout_1_2.setOnClickListener(v -> {
+            setContentView(R.layout.code_challenge_2_2);
+
+            next_layout_2_3 = findViewById(R.id.button_next_layout_2_3);
+
+            initNextLayoutButton23();
+        });
+    }
+
+    protected void initSpinner() {
+        androidVersionsList = new ArrayList<>();
```

Comment:
I think I already mentioned this idea. Please consider implementing small methods that have only one responsibility and only one reason to change